src/main.rs

```diff
@@ -142,7 +176,7 @@ fn acceptor() -> Result<TlsAcceptor> {
 }
 
 /// Return the URL requested by the client.
-async fn parse_request(stream: &mut TlsStream<TcpStream>) -> std::result::Result<Url, (u8, &'static str)> {
+async fn parse_request(handle: &mut RequestHandle) -> std::result::Result<Url, (u8, &'static str)> {
```

Comment:
Perhaps all the functions that take a `RequestHandle` argument should become methods of `RequestHandle`.

Reply:
Should `handle_request` be a method of `RequestHandle` too?